nucypher/blockchain/eth/interfaces.py

```diff
@@ -541,9 +542,7 @@ def sign_and_broadcast_transaction(self,
         try:
             txhash = self.client.send_raw_transaction(signed_raw_transaction)  # <--- BROADCAST
         except (TestTransactionFailed, ValueError) as error:
-            raise self.__transaction_failed(exception=error,
-                                            transaction_dict=transaction_dict,
-                                            transaction_name=transaction_name)
+            raise  # TODO: Unify with Transaction failed handling
```

Comment:
What needs to happen here?  What exception class needs to be raised?

Reply:
I added this comment: There is a relative new method `handle_failed_transaction` than can be used here to extract the context for easier debugging.